torch/testing/_internal/common_methods_invocations.py

```diff
@@ -3374,6 +3374,37 @@ def sample_inputs_linalg_solve(op_info, device, dtype, requires_grad=False, vect
         out.append(SampleInput(a, args=(b,)))
     return out
 
+def sample_inputs_linalg_solve_triangular(op_info, device, dtype, requires_grad=False, **kwargs):
+    make_arg = partial(make_tensor, dtype=dtype, device=device)
+    bs = (1, 2, 0)
+    ns = (3, 0)
+    ks = (1, 3, 0)
+
+    def gen_inputs():
+        for b, n, k in itertools.chain(product(bs, ns, ks)):
+            for left, upper, uni in product((True, False), repeat=3):
+                with torch.no_grad():
+                    if b == 1:
+                        A = make_arg((n, n)) if left else make_arg((k, k))
+                        B = make_arg((n, k))
+                    else:
+                        A = make_arg((b, n, n)) if left else make_arg((b, k, k))
+                        B = make_arg((b, n, k))
+                    if uni:
+                        # Not really necessary, but writing it for consistency
+                        A.diagonal(0, -2, -1).fill_(1.)
+                    else:
+                        d = A.diagonal(0, -2, -1)
+                        d[d.abs() < 1e-6] = 1.
+                    if upper:
+                        A.triu_()
+                    else:
+                        A.tril_()
+                A.requires_grad_(requires_grad)
+                B.requires_grad_(requires_grad)
```

Comment:
Could you please test all the cases when `A.requires_grad and B.requries_grad == False`.

Reply:
On a second thought, I don't know how to do this really. 
Any pointers @pmeier @krshrimali ?

Reply:
@nikitaved What would be the point? Are there different code paths used for such a scenario? Otherwise you can simply test both at the same time.

Reply:
There are branches in the code that depend on whether inputs require grads. @Lezcano, you could have a look at how, for example, OpInfo for `lu_solve `is doing that.

Reply:
Fixed. Thanks for the pointers!